examples/src/main/scala/io/finch/todo/Main.scala

```diff
@@ -31,21 +32,28 @@ object Main extends TwitterServer {
 
   private val port: Flag[Int] = flag("port", 8081, "TCP port for HTTP server")
 
-  def main(): Unit = {
-    println(s"Open your browser at http://localhost:${port()}/todo/index.html") // scalastyle:ignore
+  private def serve(service: Service[Request, Response]): Resource[IO, ListeningServer] =
+    Resource.make(for {
+      srv <- IO(Http.server.withStatsReceiver(statsReceiver))
+      server <- IO(srv.serve(s":${port()}", service))
+    } yield server) { server =>
+      IO.defer(implicitly[ToAsync[Future, IO]].apply(server.close()))
+    }
 
-    val server = for {
-      id <- Ref[IO].of(0)
-      store <- Ref[IO].of(Map.empty[Int, Todo])
-    } yield {
-      val app = new App(id, store)(IO.contextShift(ExecutionContext.global))
-      val srv = Http.server.withStatsReceiver(statsReceiver)
+  val run: IO[Unit] =
+    (for {
+      _ <- Resource.eval(IO.println(s"Open your browser at http://localhost:${port()}/todo/index.html")) // scalastyle:ignore
+      id <- Resource.eval(Ref[IO].of(0))
+      store <- Resource.eval(Ref[IO].of(Map.empty[Int, Todo]))
+      app = new App(id, store)
+      service <- app.toService
+      server <- serve(service)
+    } yield server).use(_ => IO(Await.ready(adminHttpServer)) >> IO.never)
 
-      srv.serve(s":${port()}", app.toService)
+  def main(): Unit = {
+    val dispatcher = new Dispatcher[IO] {
```

Comment:
Can we do this instead?

```
val run = Dispatcher[IO].use { dispathcer =>
  // run on the things
}

// since run is IO
run.unsafeRunSync()
```